### ISSUE TITLE
`mast.Observations.query_criteria` raises KeyError exception on empty result

### ISSUE THREAD

**Issue by barentsen** [astropy/astroquery#1044]
I encountered the exception below when attempting a MAST query that returns no results.  I reckon we'll want this to yield a more user-friendly message instead?

```
In [1]: from astroquery.mast import Observations
In [2]: obs = Observations.query_criteria(obs_collection='Kepler', objID=999999)
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-314617fc6aa5> in <module>()
----> 1 obs = Observations.query_criteria(obs_collection='Kepler', objID=212573842)

/home/gb/bin/anaconda/lib/python3.6/site-packages/astroquery/utils/class_or_instance.py in f(*args, **kwds)
     23         def f(*args, **kwds):
     24             if obj is not None:
---> 25                 return self.fn(obj, *args, **kwds)
     26             else:
     27                 return self.fn(cls, *args, **kwds)

/home/gb/bin/anaconda/lib/python3.6/site-packages/astroquery/utils/process_asyncs.py in newmethod(self, *args, **kwargs)
     27             if kwargs.get('get_query_payload') or kwargs.get('field_help'):
     28                 return response
---> 29             result = self._parse_result(response, verbose=verbose)
     30             self.table = result
     31             return result

/home/gb/bin/anaconda/lib/python3.6/site-packages/astroquery/mast/core.py in _parse_result(self, responses, verbose)
    206         for resp in responses:
    207             result = resp.json()
--> 208             resTable = _mashup_json_to_table(result)
    209             resultList.append(resTable)
    210 

/home/gb/bin/anaconda/lib/python3.6/site-packages/astroquery/mast/core.py in _mashup_json_to_table(json_obj)
     81 
     82     if not (json_obj.get('fields') and json_obj.get('data')):
---> 83         raise KeyError("Missing required key(s) 'data' and/or 'fields.'")
     84 
     85     for col, atype in [(x['name'], x['type']) for x in json_obj['fields']]:

KeyError: "Missing required key(s) 'data' and/or 'fields.'"
```

**Comment by barentsen** [comment]
(pinging @ceb8)

**Comment by keflavich** [comment]
Interesting, I get this:
```
In [2]: obs = Observations.query_criteria(obs_collection='Kepler', objID=999999)
WARNING: NoResultsWarning: Query returned no results. [astroquery.mast.core]
```
What version of astroquery are you on?  Perhaps it's fixed in the latest dev.

**Comment by ceb8** [comment]
@barentsen  I believe you are seeing a bug that I fixed in the most recent release, try updating your astroquery?

**Comment by barentsen** [comment]
My apologies!  I was convinced I was running on master, but forgot to `setup.py develop`.  I confirm it's fine on master.  Closing!

**Comment by bsipocz** [comment]
The fix was released in 0.3.7 about 2 weeks ago, so please check whether you had that version installed or an older one.
